apps/pay/components/ParsePOSPayment/index.tsx

```diff
@@ -108,20 +103,17 @@ function ParsePayment({ defaultWalletCurrency, walletId, dispatch, state }: Prop
       username: initialUsername,
     }
     if (initialQuery !== newQuery) {
-      router.push(
-        {
-          pathname: `${username}`,
-          query: {
-            amount: initialAmount,
-            sats: initialSats,
-            unit: initialUnit,
-            memo: memo ?? "",
-            display: initialDisplay,
-          },
-        },
-        undefined,
-        { shallow: true },
-      )
+      const newUrl = `${username}?${new URLSearchParams({
+        amount: initialAmount,
+        sats: initialSats,
+        unit: initialUnit,
+        memo: memo ?? "",
+        display: initialDisplay,
+        currency: defaultWalletCurrency,
+      }).toString()}`
```

Comment:
I think generally we should use the `URL` api for those type of transformation instead of crafting the url manually